Sources/SmithyHTTPAPI/HTTPResponse.swift

```diff
@@ -8,43 +8,48 @@
 import protocol Smithy.ResponseMessage
 import protocol Smithy.Stream
 import enum Smithy.ByteStream
-import class Foundation.DispatchQueue
+import class Foundation.NSRecursiveLock
 
-public class HTTPResponse: HTTPURLResponse, ResponseMessage {
+public final class HTTPResponse: ResponseMessage, @unchecked Sendable {
+    private var lock = NSRecursiveLock()
 
-    public var headers: Headers
-    public var body: ByteStream
-    public var reason: String?
+    private var _headers: Headers
+    public var headers: Headers {
+        get { lock.lock(); defer { lock.unlock() }; return _headers }
+        set { lock.lock(); defer { lock.unlock() }; self._headers = newValue }
+    }
+
+    private var _body: ByteStream
+    public var body: ByteStream {
+        get { lock.lock(); defer { lock.unlock() }; return _body }
+        set { lock.lock(); defer { lock.unlock() }; self._body = newValue }
+    }
 
     private var _statusCode: HTTPStatusCode
-    private let statusCodeQueue = DispatchQueue(label: "statusCodeSerialQueue")
     public var statusCode: HTTPStatusCode {
-        get {
-            statusCodeQueue.sync {
-                return _statusCode
-            }
-        }
-        set {
-            statusCodeQueue.sync {
-                self._statusCode = newValue
-            }
-        }
+        get { lock.lock(); defer { lock.unlock() }; return _statusCode }
+        set { lock.lock(); defer { lock.unlock() }; self._statusCode = newValue }
     }
 
+    public let reason: String?
```

Comment:
The `reason` property is never used, I do not know what the intended use is for it.
It has been converted to read-only.